avalon/tools/workfiles/app.py

```diff
@@ -439,27 +588,225 @@ def on_browse_pressed(self):
+
+        self.widgets = widgets
+        self.refresh()
+
+        # Force focus on the open button by default, required for Houdini.
+        self.widgets["files"].widgets["open"].setFocus()
+
+        self.resize(900, 600)
+
+    def on_task_changed(self):
+        # Since we query the disk give it slightly more delay
+        tools_lib.schedule(self._on_task_changed, 100, channel="mongo")
+
+    def on_asset_changed(self):
+        tools_lib.schedule(self._on_asset_changed, 50, channel="mongo")
+
+    def set_context(self, context):
+
+        if "asset" in context:
+            asset = context["asset"]
+            asset_document = io.find_one({"name": asset,
+                                          "type": "asset"})
+
+            # Set silo
+            silo = asset_document["data"].get("silo")
+            if self.widgets["assets"].get_current_silo() != silo:
+                self.widgets["assets"].set_silo(silo)
+
+            # Select the asset
+            self.widgets["assets"].select_assets([asset], expand=True)
+
+            # Force a refresh on Tasks?
+            self.widgets["tasks"].set_asset(asset_id=asset_document["_id"])
+
+        if "task" in context:
+            self.widgets["tasks"].select_task(context["task"])
+
+    def refresh(self):
+
+        # Refresh asset widget
+        self.widgets["assets"].refresh()
+
+        # Refresh breadcrumbs
+        #self.widgets["header"].set_session(session)
+
+        self._on_task_changed()
+
+    def _on_asset_changed(self):
+        asset = self.widgets["assets"].get_active_asset()
+
+        if not asset:
+            # Force disable the other widgets if no
+            # active selection
+            self.widgets["tasks"].setEnabled(False)
+            self.widgets["files"].setEnabled(False)
+        else:
+            self.widgets["tasks"].setEnabled(True)
+
+        self.widgets["tasks"].set_asset(asset)
+
+    def on_task_pressed(self):
+
+        asset = self.widgets["assets"].get_active_asset_document()
+        if not asset:
+            log.warning("No asset selected.")
+            return
+
+        task_name = self.widgets["tasks"].get_current_task()
+        if not task_name:
+            log.warning("No task selected.")
+            return
+
+        asset_name = asset["name"]
+        api.update_current_task(task=task_name, asset=asset_name)
+
+        self.refresh()
+
+    def _on_task_changed(self):
+
+        asset = self.widgets["assets"].get_active_asset_document()
+        task = self.widgets["tasks"].get_current_task()
+
+        self.widgets["tasks"].setEnabled(bool(asset))
+        self.widgets["files"].setEnabled(all([bool(task), bool(asset)]))
+
+        files = self.widgets["files"]
+        files.set_asset_task(asset, task)
+        files.refresh()
+
+
+def show(root=None, debug=False, parent=None, use_context=True):
     """Show Work Files GUI"""
+    # todo: remove `root` argument to show()
```

Comment:
Since now the Work Files tool can browse to different assets and tasks setting an explicit root makes no sense, since the root is computed based on asset/task work root.

As such, we can remove the `root` argument. Should we preserve this for API compatibility?

Reply:
Just pop a warning if `root` in `kwargs` ?